### PR TITLE
SAK-48719 grader: reduce scroll fatigue on mobile

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-48719

### CODE REVIEW

**Comment by adrianfish** [comment]
Screenshot 2023-03-31 at 19 39 13
Screenshot 2023-03-31 at 19 39 02
Screenshot 2023-03-31 at 15 45 39
Screenshot 2023-03-31 at 15 45 21
Screenshot 2023-03-31 at 15 44 55
Screenshot 2023-03-31 at 15 44 44
Screenshot 2023-03-31 at 15 44 28
Screenshot 2023-03-31 at 15 44 10
Screenshot 2023-03-31 at 15 43 44
Screenshot 2023-03-31 at 15 43 17

**Comment by adrianfish** [comment]
Screenshot 2023-03-31 at 19 39 13
Screenshot 2023-03-31 at 19 39 02
Screenshot 2023-03-31 at 15 45 39
Screenshot 2023-03-31 at 15 45 21
Screenshot 2023-03-31 at 15 44 55
Screenshot 2023-03-31 at 15 44 44
Screenshot 2023-03-31 at 15 44 28
Screenshot 2023-03-31 at 15 44 10
Screenshot 2023-03-31 at 15 43 44
Screenshot 2023-03-31 at 15 43 17

**Comment by adrianfish** [comment]
Screenshot 2023-03-31 at 15 44 28

**Comment by adrianfish** [comment]
It's worth reading the jira ticket for this PR :)

**Review by kunaljaykam — COMMENTED** [review]
Great work @adrianfish, I am leaving some suggestions.

File: webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js [hunk 1]
```diff
@@ -530,263 +454,298 @@ export class SakaiGrader extends gradableDataMixin(SakaiElement) {
...
+                          class="grader-originality-link">
+                        <i class="${result[Submission.originalityConstants.originalityIcon]}"></i>
+                        <span>${this.assignmentsI18n["content_review.disclosure.pending"]}</span>
+                      </a>
+                      <span>
+                        <span class="grader-originality-delimiter">${this.assignmentsI18n["content_review.delimiter"]}</span>
+                        <span>${result[Submission.originalityConstants.originalityName]}</span>
+                      </span>
+                      <div class="collapse grader-originality-caption" id="${result[Submission.originalityConstants.originalityKey]}">
+                        <div>
```
[the remark this answers is not in the document]
Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js:494]: I don't think so. I've removed it. Nice catch.

File: webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js [hunk 2]
```diff
@@ -530,263 +454,298 @@ export class SakaiGrader extends gradableDataMixin(SakaiElement) {
...
+                    ` : html`
+                      <a href="#${result[Submission.originalityConstants.originalityKey]}"
+                          data-bs-toggle="collapse"
+                          role="button"
+                          aria-expanded="false"
+                          aria-controls="${result[Submission.originalityConstants.originalityKey]}"
+                          class="grader-originality-link">
+                        <i class="${result[Submission.originalityConstants.originalityIcon]}"></i><span>${this.assignmentsI18n["content_review.disclosure.error"]}</span>
+                      </a>
+                      <span>
```
[the remark this answers is not in the document]
Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js:508]: I'm not touching that stuff. I didn't add it and you have to enable turnitin to test it. I don't want to break it.

File: webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js [hunk 3]
```diff
@@ -807,15 +776,46 @@ export class SakaiGrader extends gradableDataMixin(SakaiElement) {
       ${this.areSettingsInAction() ? html`
       <div class="sak-banner-warn">${this.i18n.filter_settings_warning}</div>
       ` : ""}
-      ${this.renderNav()}
+      ${this.renderTopbar()}
+      <div id="grader-submitted-block" class="grader-block">
+        <div class="d-flex mb-3">
+          <sakai-user-photo user-id="${this._getPhotoUserId()}" classes="grader-photo" profile-popup="on"></sakai-user-photo>
+          <div style="flex: 4;">
```
Comment (kunaljaykam) [inline webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js:783]:
```suggestion
          <div class="d-flex flex-grow-1">
```

**Review by kunaljaykam — APPROVED** [review]
Great work @adrianfish, Let's merge this!

File: webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js [hunk 4]
```diff
@@ -386,407 +404,358 @@ export class SakaiGrader extends gradableDataMixin(SakaiElement) {
...
-            <span>${this.assignmentsI18n["gen.assign.spent"]}</span>
-            <span> ${this.submission.submitters[0].timeSpent}</span>
-          ` : ""}
-        </div>
-      </div> <!-- /grader-submitted-block -->
+      <aside id="grader" class="offcanvas offcanvas-end" tabindex="-1" aria-labelledby="grader-label">
 
-      <div id="grader" class="${this.graderOnLeft ? "on-left" : ""}">
+        <div class="offcanvas-header">
+          <h2 class="offcanvas-title" id="grader-label">Grader</h2>
```
[the remark this answers is not in the document]
Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js:410]: yep, good spot

**Pull request merged**